File: Binary Tree Preorder Traversal/Binary Tree Preorder Traversal.py
# Definition for a  binary tree node
# class TreeNode:
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None

class Solution:
    # @param root, a tree node
    # @return a list of integers
    def preorderTraversal(self, root):
        stack=[None]
        preorder=[]
        temp=root
        while temp!=None:
            preorder.append(temp.val)
            if temp.right!=None:
                stack.append(temp.right)
            if temp.left!=None:
                stack.append(temp.left)
            temp=stack.pop()
        return preorder

File: Binary Tree Preorder Traversal/Binary Tree Preorder Traversal.java
/**
 * Definition for binary tree
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode(int x) { val = x; }
 * }
 */
public class Solution {
    public ArrayList<Integer> preorderTraversal(TreeNode root) {
        ArrayList<Integer> list=new ArrayList<Integer>();
        Stack<TreeNode> stack=new Stack<TreeNode>();
        stack.push(null);
        TreeNode temp=root;
        while(temp!=null){
            list.add(temp.val);
            if(temp.right!=null){
                stack.push(temp.right);
            }
            if(temp.left!=null){
                stack.push(temp.left);
            }
            temp=stack.pop();
        }
        return list;
    }
}
